tests/test_commenter.py

```diff
@@ -0,0 +1,466 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: MIT
+from __future__ import annotations
+
+import logging
+from argparse import Namespace
+from typing import Any, Callable
+from unittest.mock import MagicMock, Mock
+
+import pytest
+from pytest_mock import MockerFixture
+
+from openqabot.commenter import Commenter
+from openqabot.errors import NoResultsError
+from openqabot.types import ArchVer
+from openqabot.types.incident import Incident
+
+
+@pytest.fixture
+def make_job() -> Callable[..., dict[str, str | int]]:
+    def _make_job(**overrides: Any) -> dict[str, str | int]:
+        defaults = {
+            "job_id": 1,
+            "name": "test_job",
+            "status": "passed",
+            "job_group": "foo",
+            "flavor": "test-flavor",
+            "version": "something",
+            "group_id": 42,
+            "distri": "slowroll",
+            "build": "12.34",
+        }
+        return {**defaults, **overrides}
+
+    return _make_job
+
+
+@pytest.fixture
+def make_comment_api() -> Callable[..., MagicMock]:
+    def _make_comment_api(
+        comments: list | None = None,
+        comment_find_results: list[tuple] | None = None,
+        marker: str = "comment 1\ncomment 2\ncomment 3",
+    ) -> MagicMock:
+        mock = MagicMock()
+        mock.get_comments.return_value = comments
+        mock.add_marker.return_value = marker
+        mock.truncate.return_value = marker
+        mock.comment_find.side_effect = comment_find_results or [(None, None)]
+        return mock
+
+    return _make_comment_api
+
+
+@pytest.fixture
+def mock_args() -> Namespace:
+    return Namespace(dry=True, token="test_token")  # noqa: S106
+
+
+@pytest.fixture
+def mock_incident_smelt() -> Mock:
+    mock_incident = Mock(spec=Incident)
+    mock_incident.id = 1
+    mock_incident.type = "smelt"
+    mock_incident.revisions = {}
+    return mock_incident
+
+
+@pytest.fixture
+def mock_incident_smelt_with_revisions() -> Mock:
+    mock_incident = Mock(spec=Incident)
+    mock_incident.id = 2
+    mock_incident.type = "smelt"
+    mock_incident.revisions = {
+        ArchVer("x86_64", "15-SP4"): 12345,
+        ArchVer("aarch64", "15-SP4"): 12346,
+    }
+    return mock_incident
+
+
+@pytest.fixture
+def mock_commentapi() -> Mock:
```

Comment:
Is this function an unused leftover? I'm asking because the coverage report suggests that and quick search didn't reveal any matches.